wfe/web-front-end.go

```diff
@@ -316,6 +317,38 @@ const (
 	malformedJWS = "Unable to read/verify body"
 )
 
+func algorithmForKey(key *jose.JsonWebKey) (string, error) {
+	// TODO(https://github.com/letsencrypt/boulder/issues/792): Support EC.
+	switch key.Key.(type) {
+	case *rsa.PublicKey:
+		return "RS256", nil
+	}
+	return "", core.SignatureValidationError("no signature algorithms suitable for given key type")
+}
+
+// Check that (1) there is a suitable algorithm for the provided key based on its
+// Golang type, (2) the Algorithm field on the JWK is either absent, or matches
+// that algorithm, and (3) the Algorithm field on the JWK is present and matches
+// that algorithm.
+func checkAlgorithm(key *jose.JsonWebKey, parsedJws *jose.JsonWebSignature) (string, error) {
```

Comment:
Document the return values. Make constants for the stat names and for the RS256 one and document them.